backend/src/firebase/user/user.service.ts

```diff
@@ -0,0 +1,178 @@
+/* eslint-disable prettier/prettier */
+import { Injectable } from '@nestjs/common';
+import { AdminService } from '../admin/admin.service';
+import { User } from 'src/interfaces/user.interface';
+import * as bcrypt from 'bcryptjs'
+import { JwtService } from '@nestjs/jwt';
+import { Request } from 'express';
+
+@Injectable()
+export class UserService {
+  constructor(private readonly adminService: AdminService, private readonly jwtService: JwtService){}
+
+  async registerUser(data: User){
+    const firestore = this.adminService.getFirestore();
+
+    try{
+      const username = await firestore.collection('users').where('username', '==', data.username).get()
+      if(!username.empty)
+        return { data: { success: {message: ''}, error: { message: "This username is already in use" }}}
+
+      const email = await firestore.collection('users').where('email', '==', data.email).get()
+      if(!email.empty)
+        return { data: { success: {message: ''}, error: { message: "This email is already in use" } }}
+
+      if(data.rePassword !== data.password)
+        return { data: { success: {message: ''}, error: { message: "Password and Repeat password is not match" }}}
+      
+      const password = await bcrypt.hash(data.password, 10)
```

Comment:
Ta funkcja jest mega długa. Postaraj się rozbić ją na mniejsze funkcje. Np. to hashowanie hasła, albo tworzenie usera możnaby wyrzucić do osobnej metody, żeby było bardziej przejrzyste co się dzieje w tej funkcji i żeby nie miała ona tak dużo linii.